server/src/main/java/com/aws/greengrass/cli/CLIService.java

```diff
@@ -221,32 +225,28 @@ private synchronized void generateCliIpcInfo() throws UnauthenticatedException,
         Path authTokenDir = kernel.getNucleusPaths().cliIpcInfoPath();
         revokeOutdatedAuthTokens(authTokenDir);
 
-        // [P41372857]: Support Windows group permissions
-        if (Exec.isWindows) {
+        Topic authorizedGroups = config.find(CONFIGURATION_CONFIG_KEY,
+                Exec.isWindows ? AUTHORIZED_WINDOWS_GROUPS : AUTHORIZED_POSIX_GROUPS);
+        if (authorizedGroups == null) {
             generateCliIpcInfoForEffectiveUser(authTokenDir);
             return;
         }
 
-        Topic authorizedPosixGroups = config.find(CONFIGURATION_CONFIG_KEY, AUTHORIZED_POSIX_GROUPS);
-        if (authorizedPosixGroups == null) {
-            generateCliIpcInfoForEffectiveUser(authTokenDir);
-            return;
-        }
-        String posixGroups = Coerce.toString(authorizedPosixGroups);
-        if (posixGroups == null || posixGroups.length() == 0) {
+        String groups = Coerce.toString(authorizedGroups);
+        if (groups == null || groups.length() == 0) {
             generateCliIpcInfoForEffectiveUser(authTokenDir);
             return;
         }
-        for (String posixGroup : posixGroups.split(",")) {
-            posixGroup = posixGroup.trim();
-            UserPlatform.BasicAttributes group;
+        for (String group : groups.split(",")) {
+            group = group.trim();
+            UserPlatform.BasicAttributes groupAttributes;
             try {
-                group = getGroup(posixGroup);
+                groupAttributes = getGroup(group);
```

Comment:
getGroup works correctly on windows groups?

Reply:
Yes it does.